docs/fomega/deep-isorecursive/alternatives.md

```diff
@@ -16,19 +16,116 @@ Also iterated application is common at the type level as it simplifies unificati
 
 There is a [formalization sketch of this](https://gist.github.com/effectfully/8ae112e2a99393c493642fc52aafe87f#file-system-f-iso-agda).
 
+Here are some examples:
+
+the `nat` data type:
+
+```
+natF = \(nat :: *) -> all r. r -> (nat -> r) -> r
+nat  = fix natF []
+zero = wrap natF [] /\(r :: *) -> \(z : r) (f : nat -> r) -> z
+succ = \(n : nat) -> wrap natF [] /\(r :: *) -> \(z : r) (f : nat -> r) -> f n
+```
+
+the `list` data type
+
+```
+listF = \(list :: * -> *) (a :: *) -> all (r :: *). r -> (a -> list a -> r) -> r
+list  = \(a :: *) -> fix listF [a]
+nil   = /\(a :: *) -> wrap listF [a] /\(r :: *) -> \(z : r) (f : a -> list a -> r) -> z
+cons  = /\(a :: *) -> \(x : a) (xs : list a) ->
```

Comment:
What's the syntax here? I've never seen anyone use the `/\`. Do you have any references to the literature?

Reply:
This is a fairly standard notation. See for example the "16.1 Polymorphic Abstraction" chapter of [Practical Foundations for Programming Languages](https://www.cs.cmu.edu/~rwh/pfpl/2nded.pdf). `/\` is the ASCII version of `Λ`, a big lambda. People sometimes call type abstractions big lambdas (and I do this all the time).

Reply:
Yeah, this is standard AFAIK. GHC Core uses it too, it's been used by every version of System F I've seen.

Reply:
I've just checked, TAPL uses `λ` for both the kinds of lambdas.